pkg/api/secretgenerator/secretgenerator.go

```diff
@@ -0,0 +1,147 @@
+package secretgenerator
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/getlantern/deepcopy"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/util/gzip"
+)
+
+func LoadConfigFromPath(path string) (Config, error) {
+	var config Config
+	cfgBytes, err := gzip.ReadFileMaybeGZIP(path)
+	if err != nil {
+		return nil, err
+	}
+	if err := yaml.Unmarshal(cfgBytes, &config); err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+type Config []SecretItem
+
+func (c *Config) UnmarshalJSON(data []byte) error {
+	var config []SecretItem
+	if err := yaml.Unmarshal(data, &config); err != nil {
```

Comment:
```suggestion
	if err := json.Unmarshal(data, &config); err != nil {
```

Reply:
the data are in YAML format.

Reply:
No, data that is given to an UnmarshalJSON must never be in yaml format

Reply:
The yaml unmarshaler converts it to json and then calls the UnmarshalJSON IIRC

Reply:
Yea that's why it's working now.  If we need to replace to `json.Unmarshal` we will have to convert the data to JSON before passing it here. But yaml already does that.

Reply:
Or I can guess that the result will be the same

Reply:
I just checked. Changing that to `json`  generates an error.

Reply:
Then you are doing sth wrong, this definitely works, we have examples of this in our codebase like e.G.: https://github.com/openshift/ci-tools/blob/9f710b04bc1319b5374ea7ef26a8dd1e8dd97f8b/pkg/api/secretbootstrap/secretboostrap.go#L90-L95

where we implement a custom json unmarshaler that is used withing the yaml unmarshaling

Reply:
Yea that makes sense. Let me check this and update the PR. Thanks

Reply:
addressed